### DISCUSSION TITLE
SQL Invalid Parameter Number when using string IDs

### DISCUSSION THREAD

**Discussion by matthewjumpsoffbuildings** [laravel/nova-issues#5150] [Q&A] [answered]
I am using an SQL View as a model. The View generates two columns, `count` and `id`, but `id` is a unique string, not an integer.

My Eloquent Model has the following properties set to accomodate this
```php
protected $primaryKey = 'id';
protected $keyType = 'string';
public $incrementing = false;
```

My Nova resource has the following two fields
```php
ID::make(),
Number::make('Count')->sortable(),
```

The index page renders normally, but when I attempt to access the detail page, I get the following error:
```
 Invalid parameter number (SQL: select '189-274-288-294-339-480-642-685-840-955' as id limit 1)",
```
It appears to not be selecting the 'count' column? Thats my best guess as to why the parameter number would be incorrect but honestly I am not sure.

Any suggestions?

**Comment by matthewjumpsoffbuildings** [comment, marked as answer]
Turns out the issue was due to outdated detail query code from the Nova 3 to 4 upgrade...

**Reply by annenootje** [reply]
Hi Matthew,
How did you fix it?

**Reply by matthewjumpsoffbuildings** [reply]
I had overridden the `public static function detailQuery()` method on my Resource class, which I simply removed completely. Once the Resource was back to using the standard `detailQuery()` method it just worked
